src/layers/tile/mod.rs

```diff
@@ -40,8 +49,11 @@ impl LayerTileData {
     /// Use [`LayerTile::get_tile`] if you want to obtain the [`Tile`] that this layer tile is
     /// referencing.
     #[inline]
-    pub fn tileset_index(&self) -> usize {
-        self.tileset_index
+    pub fn tileset_index(&self) -> Option<usize> {
```

Comment:
This should be `tileset_location` now that the index has been replaced by location, and it should return `TilesetLocation` (should just be a readonly property)